### DISCUSSION TITLE
Different query interval for weather and weather warnings

### DISCUSSION THREAD

**Discussion by toas-koas** [breezy-weather/breezy-weather#1550] [Ideas]
### Describe your idea

Ich würde es bevorzugen wenn man den Abfrage Intervall für Wetter und Wetterwarnungen unterschiedlich einstellen könnte. 

### Other details

For weather warnings, it is necessary to be notified quickly.
The weather query must not be queried so often.

### Acknowledgements

- [X] I have searched the [existing issues](https://github.com/breezy-weather/breezy-weather/issues) and [existing discussions](https://github.com/breezy-weather/breezy-weather/discussions/categories/ideas) and this is a new ticket, **NOT** a duplicate or related to another open OR closed issue or discussion.
- [X] The idea I’m suggesting is not in the [list of features that will not be implemented](https://github.com/breezy-weather/breezy-weather/blob/main/README.md#features-that-will-not-be-implemented).
- [X] I have updated the app to **[version 5.3.1 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)**. Even if I’m getting updates from F-Droid default repo, I don’t need to uninstall/reinstall to update.
- [X] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance). I can write in my native language under the English text, if necessary.
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by papjul** [comment]
Which alert source are you using? Because alerts are usually emitted the day before, not when it happens, so you should be notified well in advance.

Unless you're talking about seismic activity, or tsunamis, or things that can't be forecasted. In that case, I recommand you use an app with a push service (we don't have it and we can't technically have it).

**Comment by toas-koas** [comment]
I use the German weather service DWD. Warnings are not always known the day before, as thunderstorms often occur spontaneously.
Other apps such as https://codeberg.org/Starfish/TinyWeatherForecastGermany have exactly this function.
There I can set the interval for severe weather to 15 minutes, for example.

**Comment by chunshek** [comment]
Weather warnings being issued days in advance is a mostly European thing. In many other places outside of Europe, severe weather warnings are often issued or canceled on an ad hoc basis.

For instance, in tropical or subtropical climates, rain is often convective rather than stratiform. Thus the timing, location, or amount of heavy rainfall often cannot be predicted beyond a few hours or for more than a few square kilometers. Warnings can only be issued ad hoc. One cannot assume to know about warnings many days in advance in those places.

But for many places in mid-latitude continental Europe, the weather pattern is usually more stable. Warnings can often be issued much further ahead of time for much wider coverage areas too. Checking for warnings every few minutes would be excessive in those cases.

- Indonesia is one of the few places which issues both ahead-of-time (D+0, D+1, D+2) and ad hoc warnings (from now for the next 3 hours).
- Taiwan sometimes issues cell warnings for just 1 to 2 hours for very specific locations.
- In Hong Kong and Macao, precaution of potential warnings may be mentioned in the detailed bulletin, but they are never "issued" per se ahead of time.

**Reply by papjul** [reply]
I see, I thought it would be limited to earthshakes, tornados or very specific things alteady covered by [Cell Broadcast](https://en.m.wikipedia.org/wiki/Cell_Broadcast).

Having alerts more frequently refreshed is not technically easy to implement, and it's hard to know which places need them more frequently (you don't want to call the alert endpoint uselessly when you live in a stable weather place like Europe).

**Reply by chunshek** [reply]
To give more background, an [article](https://www.weather.gov.sg/forecasting-2/) published by the Meteorological Service Singapore outlines the technical challenges of weather forecasting in the tropics – probably the main reason why MSS does not issue hourly forecasts which can be then used in Breezy Weather.

> Tropical weather is difficult to forecast compared to mid-latitude weather. Unlike the mid-latitudes, the tropics have a relatively homogenous air mass and fairly uniform distribution of surface temperature and air pressure. Therefore local and mesoscale effects (e.g. [sea breezes](https://www.weather.gov.sg/learn_winds/#atmosphere2)) are more dominant than synoptic (large-scale) influences, except for tropical cyclones.
>
> Strong convection and moist air in the tropics gives rise to frequent development of rain showers and heavy thunderstorms. These convective weather systems tend to develop and dissipate quickly, often within one or two hours. Also, as the prevailing winds in the tropics are generally light, predicting the movement of localised storms can be difficult. Their short lifespan and small size presents a forecasting challenge in terms of accuracy and lead time.
>
> Current NWP models have low skills in predicting tropical convection, and the complex dynamical processes that influence weather and climate in the tropics is still poorly understood. This is an important area of research that is being undertaken at the [Centre for Climate Research Singapore (CCRS)](https://ccrs.weather.gov.sg/) and other research centres around the world.
